semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/span/SpanSemanticConventionUtils.java

```diff
@@ -31,4 +37,49 @@
     }
     return Optional.of(host);
   }
+
+  /**
+   * @param attributeValueMap map of span data and attribute value
+   * @return URI based on OTel format
+   */
+  public static Optional<String> getURIForOtelFormat(Map<String, AttributeValue> attributeValueMap) {
+    AttributeValue hostAttribute = attributeValueMap.getOrDefault(
+        OTEL_NET_PEER_NAME,
+        attributeValueMap.get(OTEL_NET_PEER_IP));
+    if (null == hostAttribute || StringUtils.isBlank(hostAttribute.getValue())) {
+      return Optional.empty();
+    }
+    if (attributeValueMap.containsKey(OTEL_NET_PEER_PORT)
+        && !StringUtils.isBlank(attributeValueMap.get(OTEL_NET_PEER_PORT).getValue())) {
+      return Optional.of(String.format(
+          "%s:%s", hostAttribute.getValue(),
+          attributeValueMap.get(OTEL_NET_PEER_PORT).getValue()));
+    }
+    return Optional.of(hostAttribute.getValue());
+  }
+
+  public static boolean isClientSpanForOtelFormat(Map<String, AttributeValue> attributeValueMap) {
+    if (attributeValueMap.containsKey(OTelSpanSemanticConventions.SPAN_KIND.getValue())) {
+      return OTelSpanSemanticConventions.SPAN_KIND_CLIENT_VALUE.getValue().equals(
+          attributeValueMap.get(OTelSpanSemanticConventions.SPAN_KIND.getValue()).getValue());
+    }
+    return false;
+  }
+
+  public static boolean isServerSpanForOtelFormat(Map<String, AttributeValue> attributeValueMap) {
+    if (attributeValueMap.containsKey(OTelSpanSemanticConventions.SPAN_KIND.getValue())) {
+      return OTelSpanSemanticConventions.SPAN_KIND_SERVER_VALUE.getValue().equals(
+          attributeValueMap.get(OTelSpanSemanticConventions.SPAN_KIND.getValue()).getValue());
+    }
+    return false;
+  }
+
+  public static boolean isValidUri(String uri) {
+    try {
+      new URI(uri);
```

Comment:
Do we need URI or URL? URI is a superset of URL. This can introduce an issue. Check httpFieldGenrato's getNormalizedUrl to get `absolute` or `relative` URL.

Reply:
Sure let me fix this in a subsequent pr